tests/regression/test_interpolate_cross_mesh.py

```diff
@@ -0,0 +1,181 @@
+from firedrake import *
+import numpy as np
+import pytest
+from functools import reduce
+from operator import mul
+
+
+def allgather(comm, coords):
+    """Gather all coordinates from all ranks."""
+    coords = coords.copy()
+    coords = comm.allgather(coords)
+    coords = np.concatenate(coords)
+    return coords
+
+
+@pytest.fixture(params=["unitsquare", "circlemanifold", "extrudedcube"])
+def parameters(request):
+    if request.param == "unitsquare":
+        m_src = UnitSquareMesh(2, 3)
+        m_dest = UnitSquareMesh(3, 5, quadrilateral=True)
+        coords = np.array(
+            [[0.56, 0.6], [0.1, 0.9], [0.9, 0.1], [0.9, 0.9], [0.726, 0.6584]]
+        )  # fairly arbitrary
+        # add the coordinates of the mesh vertices to test boundaries
+        vertices_src = allgather(m_src.comm, m_src.coordinates.dat.data_ro)
+        coords = np.concatenate((coords, vertices_src))
+        vertices_dest = allgather(m_dest.comm, m_dest.coordinates.dat.data_ro)
+        coords = np.concatenate((coords, vertices_dest))
+        expr_op = mul
+    elif request.param == "circlemanifold":
+        m_src = UnitSquareMesh(2, 3)
+        # shift to cover the whole circle
+        m_src.coordinates.dat.data[:] *= 2
+        m_src.coordinates.dat.data[:] -= 1
+        m_dest = CircleManifoldMesh(1000)  # want close to actual unit circle
+        coords = np.array(
+            [
+                [0, 1],
+                [1, 0],
+                [-1, 0],
+                [0, -1],
+                [sqrt(2) / 2, sqrt(2) / 2],
+                [-sqrt(3) / 2, 1 / 2],
+            ]
+        )  # points that ought to be on the unit circle
+        # only add target mesh vertices since they are common to both meshes
+        vertices_dest = allgather(m_dest.comm, m_dest.coordinates.dat.data_ro)
+        coords = np.concatenate((coords, vertices_dest))
+        # function.at often gets conflicting answers across boundaries for this
+        # mesh, so we lower the tolerance a bit for this test
+        m_dest.tolerance = 0.5
```

Comment:
Having to do this makes me wonder if the current default tolerance of `1.0` is too high.

Reply:
Sadly `.at` doesn't have my fancy voting algorithm to fall back on!